### DISCUSSION TITLE
Auth Desktop App Password Lock?

### DISCUSSION THREAD

**Discussion by JohnDoe-JohnDoe** [ente-io/ente#1858] [Enhancements]
One the Auth Desktop (Windows) app, there doesn't appear to be any way to create a password or other such lock. The app opens without having to give any credentials. This seems very insecure. The only thing I can do in logout but then it requires an internet connection to login.

Please add some kind of lock to the app. (The android version uses the system lock which others have mentioned, doesn't work on the desktop)

**Comment by borque** [comment]
Hi there!
I think you mean something very similar to my suggestion [#1874](https://github.com/ente-io/ente/discussions/1874).
I would also like to suggest an encryption of the locally stored data. At the moment it seems to be stored in a plain json file in the home directory!

**Reply by JohnDoe-JohnDoe** [reply]
Yes, I saw your post but before you edited it, it only seemed to be referring to the mobile version but I see that as the desktop app is essentially a copy of the android version then what you suggest would also fix the issue.
Very concerned that the data is stored without any encryption. I'll stop using it until that is changed.

**Comment by daithi-coyle** [comment]
+1 to this and #1874. Current behaviour on Windows is to open the app, unlocked with all 2fa visible upon login. Seems very insecure.
Windows Hello integration (fingerprint) would be ideal but even a pin would be better than nothing.

**Comment by Gobias64** [comment]
+1 There's the same issue for Macs without Touch ID.

We need some options for locking the desktop app. Any (or all) of these would be nice options to have:
- Ente Auth password
- Ente Auth passkey
- Unique app password
- Unique app PIN (ideally as many digits as we want)
- Device user login password (this is the fallback for Touch ID anyway)
- Windows Hello (for devices that support it)

**Comment by sv3nnie** [comment]
This is now available in [v3.1.1](https://github.com/ente-io/ente/releases/tag/auth-v3.1.1) (see changelog of v3.1.0). You can enable it through settings -> Security -> App lock

**Reply by Gobias64** [reply]
This isn't working in 3.1.1 on a Mac without TouchID. I still get a "No system lock found" error.
![image](https://github.com/user-attachments/assets/b9628a89-82a6-4717-ae57-83de2e04c4a8)

**Reply by PyMustafa** [reply]
+1, same with me

**Reply by ashilkn** [reply]
@Gobias64 @PyMustafa This issue has been fixed on the [latest release](https://github.com/ente-io/ente/releases/tag/auth-v3.1.3) :)

**Comment by ashilkn** [comment]
Custom lock (PIN, password, or device lock) is now available on auth's desktop and mobile apps! You can find the latest release [here](https://github.com/ente-io/ente/releases/tag/auth-v3.1.3)

**Reply by emacsomancer** [reply]
There doesn't seem to be a way to set a timeout autolock?

**Reply by Gobias64** [reply]
> There doesn't seem to be a way to set a timeout autolock?

This has been reported/requested elsewhere (including by me). The app still needs a feature for autolocking after a set time of inactivity (like password managers have).
